packages/strapi-plugin-users-permissions/services/UsersPermissions.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+const fs = require('fs')
+const path = require('path');
+const stringify = JSON.stringify;
+const _ = require('lodash');
+// const Service = strapi.plugins['users-permissions'].services;
+/**
+ * UsersPermissions.js service
+ *
+ * @description: A set of functions similar to controller's actions to avoid code duplication.
+ */
+
+module.exports = {
+  getActions: () => {
+    const generateActions = (data) => (
+      Object.keys(data).reduce((acc, key) => {
+        acc[key] = { enabled: false, policy: '' };
+
+        return acc;
+    }, {}));
+
+    const appControllers = Object.keys(strapi.api || {}).reduce((acc, key) => {
+      acc.controllers[key] = generateActions(strapi.api[key].controllers[key]);
+
+      return acc;
+    }, { controllers: {} });
+
+    const pluginsPermissions = Object.keys(strapi.plugins).reduce((acc, key) => {
+      const pluginControllers = Object.keys(strapi.plugins[key].controllers).reduce((obj, k) => {
+        obj.icon = strapi.plugins[key].package.strapi.icon;
+        obj.controllers[k] = generateActions(strapi.plugins[key].controllers[k]);
+
+        return obj;
+
+      }, { icon: '', controllers: {} });
+
+      acc[key] = pluginControllers;
```

Comment:
This variable `pluginControllers ` is not necessary. You can directly assign the value to `acc[key]` ;)